### DISCUSSION TITLE
Why EcmaScript private fields are not supported?

### DISCUSSION THREAD

**Discussion by OnkelTem** [mobxjs/mobx#3753] [Q&A] [answered]
from https://mobx.js.org/observable-state.html#limitations :

> 11. EcmaScript private fields (#field) are not supported. When using TypeScript, it is recommended to use the private modifier instead.

Obviously, if we register fields via `this` (as in `makeAutoObservable(this)`) all private fields are not accessible. However, isn't there another way to replace such fields with proxies?

**Comment by mweststrate** [comment, marked as answer]
Proxies work at the object level, not field level. Once the new decorator
proposal is fully settled, and we removed the legacy implementation, we
could potentially start supporting private fields.

**Reply by OnkelTem** [reply]
Thank you, Michel!
